fmriprep/workflows/anatomical.py

```diff
@@ -74,43 +74,44 @@ def t1w_preprocessing(settings, name='t1w_preprocessing'):
             testing=settings.get('debug', False),
             template='mni_icbm152_nlin_asym_09c'
         ),
-        name='T1_2_MNI_Registration'
+        name='t1_2_mni'
     )
     # should not be necesssary but does not hurt - make sure the multiproc
     # scheduler knows the resource limits
     t1_2_mni.interface.num_threads = settings['ants_nthreads']
 
     # 6. FreeSurfer reconstruction
     if settings['freesurfer']:
-        surface_recon = surface_reconstruction(settings=settings)
+        surface_recon = surface_reconstruction(name='surface_recon',
```

Comment:
Right, we wouldn't want to accept arbitrary sub/superstrings. If we want to set a few guidelines, here are some:

* Remove `_wf` from the function name before applying other rules
* For substrings, the final word of the function name can be abbreviated or removed
  * Substrings can have a number or `_desc` appended
* Superstrings must be the function name followed by a number or `_desc`

Where `_desc` is an underscore followed by a one word (or abbreviation) description.

Looking at what we've got and proposed names that fit these rules:

* `skullstrip_wf`/`skullstrip_ants` => `skullstrip`
* `surface_reconstruction` => `surface_recon`
* `epi_hmc` => `epi_hmc`
* `ref_epi_t1_registration` => `ref_epi_t1_reg`
* `discover_wf` => `discover`
* `sdc_unwarp` => `sdc_unwarp`
* `fieldmap_unwarping_report` => `fieldmap_unwarping_report`
* `epi_mni_transformation` => `epi_mni`
* `epi_surf_sample` => `epi_surf`
* `cleanup_edge_pipeline` => `cleanup_edge`
* `phdiff_workflow` => `phdiff`

Not perfect, but it doesn't seem too far off.

-----

For function naming schemes, I think a reasonable rule would be for workflow-returning functions to end in `_wf` or `_workflow`. (Slight preference for `_wf`.)

I don't think we have functions that return nodes.

What about functions that are intended to be wrapped with the `Function` interface? Having a name scheme might make things easier. Supposing we ended all such functions with `_fn`:

```Python
def gifti_name_fn(arg1, arg2):
    ...
    return ret

gifti_name = pe.Node(niu.Function(function=gifti_name_fn, ...), name='gifti_name')
```

I don't really like `_fn`, since it isn't obviously for wrappable functions, but something like that could keep the namespace clean.